File: DeJia/app/src/main/java/com/dejia/anju/api/GetBuildingInfoApi.java
package com.dejia.anju.api;

import android.content.Context;

import com.dejia.anju.api.base.BaseCallBackApi;
import com.dejia.anju.api.base.BaseCallBackListener;
import com.dejia.anju.net.NetWork;

import java.util.Map;

//获取楼盘信息
public class GetBuildingInfoApi implements BaseCallBackApi {
    @Override
    public void getCallBack(Context context, Map<String, Object> maps, final BaseCallBackListener listener) {
        NetWork.getInstance().call("tour", "getBuilding", maps, mData -> listener.onSuccess(mData));
    }
}

File: DeJia/app/src/main/java/com/dejia/anju/api/GetBuildingInfoApi.kt
package com.dejia.anju.api

import android.content.Context
import com.dejia.anju.api.base.BaseCallBackApi
import com.dejia.anju.api.base.BaseCallBackListener
import com.dejia.anju.net.NetWork
import com.dejia.anju.net.ServerData

//获取楼盘信息
class GetBuildingInfoApi : BaseCallBackApi {
    override fun getCallBack(context: Context, maps: MutableMap<String, Any>, listener: BaseCallBackListener<*>) {
        NetWork.instance?.call("tour", "getBuilding", maps, context) { mData: ServerData? -> listener.onSuccess(mData) }
    }
}